src/Business Foundation/App/NoSeries/src/Batch/NoSeriesBatchImpl.Codeunit.al

```diff
@@ -0,0 +1,177 @@
+             (TempGlobalNoSeriesLine."Line No." = TempNoSeriesLine."Line No."));
+    end;
+
+    procedure PeekNextNo(NoSeriesCode: Code[20]): Code[20]
+    begin
+        exit(PeekNextNo(NoSeriesCode, WorkDate()));
+    end;
+
+    procedure PeekNextNo(NoSeriesCode: Code[20]; UsageDate: Date): Code[20]
+    var
+        TempNoSeriesLine: Record "No. Series Line" temporary;
+    begin
+        GetNoSeriesLine(TempNoSeriesLine, NoSeriesCode, UsageDate);
+        exit(PeekNextNo(TempNoSeriesLine, UsageDate));
+    end;
+
+    procedure PeekNextNo(TempNoSeriesLine: Record "No. Series Line" temporary; UsageDate: Date): Code[20]
+    var
+        NoSeries: Codeunit "No. Series";
+    begin
+        SetInitialState(TempNoSeriesLine);
+        exit(NoSeries.PeekNextNo(TempGlobalNoSeriesLine, UsageDate));
+    end;
+
+    procedure GetNextNo(NoSeriesCode: Code[20]; UsageDate: Date; HideErrorsAndWarnings: Boolean): Code[20]
+    var
+        TempNoSeriesLine: Record "No. Series Line" temporary;
+    begin
+        GetNoSeriesLine(TempNoSeriesLine, NoSeriesCode, UsageDate);
+        exit(GetNextNo(TempNoSeriesLine, UsageDate, HideErrorsAndWarnings));
+    end;
+
+    procedure GetNextNo(TempNoSeriesLine: Record "No. Series Line" temporary; UsageDate: Date; HideErrorsAndWarnings: Boolean): Code[20]
+    var
+        NoSeries: Codeunit "No. Series";
+    begin
+        SetInitialState(TempNoSeriesLine);
+        LockedNoSeriesLine.LockTable();
+        exit(NoSeries.GetNextNo(TempGlobalNoSeriesLine, UsageDate, HideErrorsAndWarnings));
+    end;
+
+    procedure SimulateGetNextNo(NoSeriesCode: Code[20]; UsageDate: Date; PrevDocumentNo: Code[20]): Code[20]
+    var
+        TempNoSeriesLine: Record "No. Series Line" temporary;
+        NoSeriesStatelessImpl: Codeunit "No. Series - Stateless Impl.";
+    begin
+        if NoSeriesCode = '' then
+            exit(IncStr(PrevDocumentNo));
+
+        SetSimulationMode();
+
+        GetNoSeriesLine(TempNoSeriesLine, NoSeriesCode, UsageDate);
+        TempNoSeriesLine."Last No. Used" := PrevDocumentNo;
+        if not NoSeriesStatelessImpl.EnsureLastNoUsedIsWithinValidRange(TempNoSeriesLine, true) then
+            exit(IncStr(PrevDocumentNo));
+
+        TempNoSeriesLine.Modify(false);
+        exit(GetNextNo(TempNoSeriesLine, UsageDate, false));
+    end;
+
+    procedure GetLastNoUsed(NoSeriesCode: Code[20]): Code[20]
+    var
+        TempNoSeriesLine: Record "No. Series Line" temporary;
+        NoSeries: Codeunit "No. Series";
+    begin
+        GetNoSeriesLine(TempNoSeriesLine, NoSeriesCode, WorkDate());
+        exit(NoSeries.GetLastNoUsed(TempGlobalNoSeriesLine));
+    end;
+
+    procedure GetLastNoUsed(TempNoSeriesLine: Record "No. Series Line" temporary): Code[20]
+    var
+        NoSeries: Codeunit "No. Series";
+    begin
+        SetInitialState(TempNoSeriesLine);
+        exit(NoSeries.GetLastNoUsed(TempGlobalNoSeriesLine));
+    end;
+
+    procedure SetSimulationMode()
```

Comment:
I don't know what the actual use is, however, would there be reason to unset SimulationMode?

Reply:
Once you are in Simulation mode, you don't want to risk writing this back to disk. It's a dangerous can of worms to open up.